From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joseph Myers <joseph@codesourcery.com>
Date: Fri, 20 Apr 2018 16:36:00 +0000
Subject: [PATCH] Update aarch64 bits/hwcap.h, dl-procinfo.c for Linux 4.16
 HWCAP_ASIMDFHM.

This patch updates the aarch64 bits/hwcap.h and dl-procinfo.c for the
new HWCAP_ASIMDFHM value in Linux 4.16.

Tested with build-many-glibcs.py for aarch64-linux-gnu.

	* sysdeps/unix/sysv/linux/aarch64/bits/hwcap.h (HWCAP_ASIMDFHM):
	New macro.
	* sysdeps/unix/sysv/linux/aarch64/dl-procinfo.c (_DL_HWCAP_COUNT):
	Increase to 24.
	(_dl_aarch64_cap_flags): Add asimdfhm.

(cherry picked from commit 458b94f63e6fc5af1582a98007a1e4769f785fb7)

diff --git a/ChangeLog b/ChangeLog
index 73243ff8fadddc5fd2d85570595ae9e2e96c1cf2..df116fb76ee91d6f30e9c423139957ab6b31e043 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -135,6 +135,14 @@
 	* resolv/tst-resolv-nondecimal.c: Likewise.
 	* sysdeps/posix/getaddrinfo.c (gaih_inet): Call __inet_aton_exact.
 
+2018-04-20  Joseph Myers  <joseph@codesourcery.com>
+
+	* sysdeps/unix/sysv/linux/aarch64/bits/hwcap.h (HWCAP_ASIMDFHM):
+	New macro.
+	* sysdeps/unix/sysv/linux/aarch64/dl-procinfo.c (_DL_HWCAP_COUNT):
+	Increase to 24.
+	(_dl_aarch64_cap_flags): Add asimdfhm.
+
 2017-11-11  Florian Weimer  <fweimer@redhat.com>
 
 	[BZ #22409]
diff --git a/sysdeps/unix/sysv/linux/aarch64/bits/hwcap.h b/sysdeps/unix/sysv/linux/aarch64/bits/hwcap.h
index 4cd37d9d76c0f45d9400de357a69cf379cb0db2a..5e5f81afea41c6f4bb645a5b690a518e0116742f 100644
--- a/sysdeps/unix/sysv/linux/aarch64/bits/hwcap.h
+++ b/sysdeps/unix/sysv/linux/aarch64/bits/hwcap.h
@@ -44,3 +44,4 @@
 #define HWCAP_ASIMDDP		(1 << 20)
 #define HWCAP_SHA512		(1 << 21)
 #define HWCAP_SVE		(1 << 22)
+#define HWCAP_ASIMDFHM		(1 << 23)
diff --git a/sysdeps/unix/sysv/linux/aarch64/dl-procinfo.c b/sysdeps/unix/sysv/linux/aarch64/dl-procinfo.c
index 252f5356bd482d9e33d89965d28ae54bab2a9c59..892a844cc3d5ae284b90835800d9efe78452b88b 100644
--- a/sysdeps/unix/sysv/linux/aarch64/dl-procinfo.c
+++ b/sysdeps/unix/sysv/linux/aarch64/dl-procinfo.c
@@ -57,7 +57,7 @@ PROCINFO_CLASS struct cpu_features _dl_aarch64_cpu_features
 #endif
 
 /* Number of HWCAP bits set.  */
-#define _DL_HWCAP_COUNT 23
+#define _DL_HWCAP_COUNT 24
 
 #if !defined PROCINFO_DECL && defined SHARED
   ._dl_aarch64_cap_flags
@@ -68,7 +68,8 @@ PROCINFO_CLASS const char _dl_aarch64_cap_flags[_DL_HWCAP_COUNT][10]
 /* Matches the names in arch/arm64/kernel/cpuinfo.c of Linux.  */
 = { "fp", "asimd", "evtstrm", "aes", "pmull", "sha1", "sha2", "crc32",
     "atomics", "fphp", "asimdhp", "cpuid", "asimdrdm", "jscvt", "fcma",
-    "lrcpc", "dcpop", "sha3", "sm3", "sm4", "asimddp", "sha512", "sve" }
+    "lrcpc", "dcpop", "sha3", "sm3", "sm4", "asimddp", "sha512", "sve",
+    "asimdfhm" }
 #endif
 #if !defined SHARED || defined PROCINFO_DECL
 ;
